core/dbt/events/types.py

```diff
@@ -109,16 +101,16 @@ def message(self) -> str:
 
 @dataclass
 class MainEncounteredError(ErrorLevel, NoFile):
-    e: BaseException
+    e: str = ''
```

Comment:
This undermines one of the major points of the event module. mypy should be able to enforce that we are providing real values for each of the expected fields. Yes mypy knows that we have a value for each of the fields we care about, but we'll never know if this default is used. And if this default is used something is very wrong.

Reply:
The problem is that we don't know what exception class is being passed in, so I thought it was safer in that case to stringify the exceptions. Which is all that we end up doing with it anyway. We can serialize a BaseException instead, but it amounts to the same thing.

Reply:
yeah this one is a special case because it really could be any subclass of `BaseException`. I still like taking a `BaseException` here because it reduces the surface area of valid `MainEncounteredError` values which makes mypy more helpful. The runtime effect is only the same if we are sure we are always stringifying an exception at every call site which mypy can enforce when structured this way.

Reply:
I changed this to be a BseException again. I'm not convinced it's an inmprovement...